test/e2e/gitops_service_test.go

```diff
@@ -247,3 +261,44 @@ func tearDownArgoCD(t *testing.T) {
 	assertNoError(t, err)
 
 }
+
+func validateMachineConfigUpdates(t *testing.T) {
+	framework.AddToFrameworkScheme(configv1.AddToScheme, &configv1.Image{})
+	ctx := framework.NewContext(t)
+	defer ctx.Cleanup()
+	f := framework.Global
+	path, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	imageYAML := filepath.Join(path, "yamls", "image.yaml")
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cmd := exec.Command(ocPath, "apply", "-f", imageYAML)
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(5 * time.Second)
+
+	if helper.ApplicationHealthStatus("image", "openshift-gitops"); err != nil {
+		t.Fatal(err)
+	}
+
+	if helper.ApplicationSyncStatus("image", "openshift-gitops"); err != nil {
+		t.Fatal(err)
+	}
+
+	existingImage := &configv1.Image{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "cluster",
+		},
+	}
+
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: existingImage.Name}, existingImage)
```

Comment:
Checking the existence of the resource is not good enough.   We should validate the CR and the status is what we expect.

Reply:
The status has been verified here - https://github.com/redhat-developer/gitops-operator/pull/127/files#diff-451b2db506ec846e24c6d56da6531ff480580cd9812c2daf225cc5a1cf7fa35cR288-R294

Reply:
I mean the CR of the openshift image.

Reply:
@wtam2018 Spend some time to get it done and even with a call with @iam-veeramalla to get more understanding.

However till now what i understood is that the test verification also should done for status - https://github.com/openshift/openshift-gitops-examples/blob/master/argocd/image/image.yaml#L13. But i do not know how to verify it.

Here is the steps followed but no luck

Steps 1:
Created and synced the image CR https://github.com/openshift/openshift-gitops-examples/blob/master/argocd/image/image.yaml successfully

Step 2: 
$ oc get image cluster -o json
Error from server (NotFound): images.image.openshift.io "cluster" not found

I was expecting Step 2 will capture the image CR status. Any further tips would really be helpful